### PR TITLE
Fix bug where pre-citation missing or no publication is not shown

### PR DESCRIPTION
**Pull request by jacobwindsor**
dcp-ops-598 dcp-683

- Fixes bug where projects that don't have `publicationsInfo` are not shown. 
- Adds test to ensure this doesn't happen again
- Refactors test
  - Spec now only contains one project
  - A `makeInputData` function is added to the test suite that uses the spec project, a mapper, and generates a list of projects
  - This is a bit easier to reason about and don't have to keep track of laods of projects in the spec JSON

### CODE REVIEW

**Review by aaclan-ebi — APPROVED** [review]

File: src/app/projects/services/projects.service.ts [hunk 1]
```diff
@@ -274,7 +274,9 @@ export class ProjectsService implements OnDestroy {
         ),
         dbgapAccessions: obj.content.dbgap_accessions ?? [],
         publications: obj.publicationsInfo ?? [],
-        authors: obj.publicationsInfo ? obj.publicationsInfo[0].authors : [],
+        authors: obj.publicationsInfo?.length
+          ? obj.publicationsInfo[0].authors || []
+          : [],
```
Comment (amnonkhen) [inline src/app/projects/services/projects.service.ts:279]: Have you tried this?

```suggestion
        authors: obj.publicationsInfo?.[0]?.authors || []
```

**Review by amnonkhen — COMMENTED** [review]
LGTM.
Well done for refactoring and simplifying the test.
1 minor suggestion

File: src/app/projects/services/projects.service.ts (hunk 1 above)
Comment (jacobwindsor) [reply src/app/projects/services/projects.service.ts:279]: That's not valid syntax

Comment (jacobwindsor) [reply src/app/projects/services/projects.service.ts:279]: I didn't realise that was valid syntax, nice

**Review by amnonkhen — DISMISSED** [review]

**Pull request merged**